spek-runtime/src/commonMain/kotlin/org/spekframework/spek2/runtime/Executor.kt

```diff
@@ -108,6 +118,15 @@ class Executor {
         }
     }
 
+    private sealed class TimedExecutionResult {
```

Comment:
Need to distinguish:
- `null` case, which means test has timed out.
- success case, where the test has succeeded.
- failure case, where the test has failed.